### PR TITLE
zh-cn: update the translation of `arguments.callee`

### PR DESCRIPTION
**Pull request by yin1999**
### Description

update the translation of `arguments.callee` & correct some translations of 'named function'

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/functions/method_definitions/index.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ var obj = {
 };
 ```
 
-> **备注：** 简写语法使用命名函数而不是匿名函数（如…`foo: function() {}`…）。命名函数可以从函数体调用（这对匿名函数是不可能的，因为没有标识符可以引用）。详细信息，请参阅{{jsxref("Operators/function","function","#Examples")}}。
+> **备注：** 简写语法使用具名函数而不是匿名函数（如…`foo: function() {}`…）。具名函数可以从函数体调用（这对匿名函数是不可能的，因为没有标识符可以引用）。详细信息，请参阅{{jsxref("Operators/function","function","#Examples")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/functions/method_definitions/index.md:62]: only modified the translation of "named function". I would update the translation later "as the whole documentation is out-of-date.

File: files/zh-cn/web/javascript/reference/functions/arguments/callee/index.md [hunk 2]
```diff
@@ -3,27 +3,33 @@ title: arguments.callee
 slug: Web/JavaScript/Reference/Functions/arguments/callee
 ---
 
-{{jsSidebar("Functions")}}
+{{jsSidebar("Functions")}}{{Deprecated_Header}}
 
-**`arguments.callee`** 属性包含当前正在执行的函数。
+> **警告：** 在[严格模式](/zh-CN/docs/Web/JavaScript/Reference/Strict_mode)下访问 `arguments.callee` 会抛出 {{jsxref("TypeError")}}。若一个函数必须引用自身，请为[函数表达式](/zh-CN/docs/Web/JavaScript/Reference/Operators/function)命名，或使用[函数声明](/zh-CN/docs/Web/JavaScript/Reference/Statements/function)。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/functions/arguments/callee/index.md:8]: Original en-US document use 'Note' here

```suggestion
> **备注：** 在[严格模式](/zh-CN/docs/Web/JavaScript/Reference/Strict_mode)下访问 `arguments.callee` 会抛出 {{jsxref("TypeError")}}。若一个函数必须引用自身，请为[函数表达式](/zh-CN/docs/Web/JavaScript/Reference/Operators/function)命名，或使用[函数声明](/zh-CN/docs/Web/JavaScript/Reference/Statements/function)。
```

File: files/zh-cn/web/javascript/reference/functions/arguments/callee/index.md [hunk 3]
```diff
@@ -33,109 +39,103 @@ function factorial(n) {
 
 ```js
 [1, 2, 3, 4, 5].map(function (n) {
-  return !(n > 1) ? 1 : /* what goes here? */ (n - 1) * n;
+  return n <= 1 ? 1 : /* 这里该怎么写？*/ (n - 1) * n;
 });
 ```
 
-这个不行。为了解决这个问题，`arguments.callee` 添加进来了。然后你可以这么做
+这个就不行。为了解决这个问题，`arguments.callee` 被添加了进来。然后你可以这么做
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/functions/arguments/callee/index.md:46]:
```suggestion
这个就不行。为了解决这个问题，`arguments.callee` 被添加了进来。然后你可以这么做：
```

File: files/zh-cn/web/javascript/reference/functions/arguments/callee/index.md [hunk 4]
```diff
@@ -33,109 +39,103 @@ function factorial(n) {
...
+
+console.log(
+  [1, 2, 3, 4, 5].map(
+    // 将高阶函数包装在 Y 组合子中
+    // “factorial”不是函数的名称：它是作为参数引入的
+    Y((factorial) => (n) => (n <= 1 ? 1 : factorial(n - 1) * n)),
+  ),
+);
+// [ 1, 2, 6, 24, 120 ]
 ```
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/functions/arguments/callee/index.md:140]: missing **Note** here

https://github.com/mdn/content/blob/c919dcbb83bc360299b21f3b35bf645b7860eb43/files/en-us/web/javascript/reference/functions/arguments/callee/index.md?plain=1#L145

**Review by jasonren0403 — COMMENTED** [review]
some suggestions

**Review by jasonren0403 — APPROVED** [review]
Thanks @yin1999 👍

**Pull request merged**